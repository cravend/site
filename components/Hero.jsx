/** @jsx jsx */
import { jsx, Container } from 'theme-ui';
import { node } from 'prop-types';

const Hero = ({ children }) => (
  <Container
    variant="wide"
    as="header"
    m={0}
    p={0}
    mb={4}
    sx={{
      backgroundColor: 'primary',
      background: theme => `${theme.background.gradient}`,
      height: 250,
      paddingBottom: [0, 37.5],
      clipPath: ['normal', 'polygon(0 0, 100% 0%, 100% 85%, 0% 100%)'],
      display: 'flex',
      alignItems: 'center',
    }}
  >
    {children}
  </Container>

);

Hero.propTypes = {
  children: node.isRequired,
};

export default Hero;
