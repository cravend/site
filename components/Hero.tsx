import React, { ReactChild, FC } from 'react';
import { Box, get, ThemeUICSSObject } from 'theme-ui';

type HeroType = {
  isDiagonal?: boolean
  children: ReactChild
}

const Hero: FC<HeroType> = ({ isDiagonal, children }: HeroType) => {
  const styles: ThemeUICSSObject = isDiagonal
    ? {
      backgroundColor: 'primary',
      background: (theme) => get(theme, 'background.gradient') as string,
      height: 250,
      paddingBottom: [0, 37.5],
      clipPath: ['normal', 'polygon(0 0, 100% 0%, 100% 85%, 0% 100%)'],
      display: 'flex',
      alignItems: 'center',
    }
    : {
      backgroundColor: 'primary',
      background: (theme) => get(theme, 'background.gradient') as string,
      height: 150,
      display: 'flex',
      alignItems: 'center',
    };

  return (
    <Box as="header" m={0} p={0} mb={[0, 4]} sx={styles}>{children}</Box>
  );
};

Hero.defaultProps = {
  isDiagonal: false,
};

export default Hero;
