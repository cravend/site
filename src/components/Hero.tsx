import styles from "../styles/modules/Hero.module.scss";
import type { ReactNode } from "react";

export type HeroProps = {
  isDiagonal?: boolean;
  children: ReactNode;
};

const Hero = ({ isDiagonal = false, children }: HeroProps) => (
  <header className={isDiagonal ? styles.diagonal : styles.normal}>
    {children}
  </header>
);
export default Hero;
