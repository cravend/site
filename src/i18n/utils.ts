import { locales, defaultLocale, RTL_LOCALES } from "./config";

import type { Locale } from "./config";
import type { NextRouter } from "next/router";

/**
 * @description This function is used to check if a string is a valid locale.
 *
 * @param localeToTest The string to check.
 * @returns True if the string is a valid locale, false otherwise.
 */
export const isLocale = (localeToTest?: string): localeToTest is Locale =>
  locales.some((locale) => locale === localeToTest);

/**
 * @description This function is used to get the current locale.
 *
 * If a custom locale is set by \@next/router, it will be returned.
 * If no custom locale is set, the router's default locale will be returned.
 * If the router's default locale isn't set, the default locale will be returned.
 *
 * @param router The NextRouter instance.
 * @returns The current locale.
 */
export const getLocale = (router: NextRouter | null) => {
  if (!router) return defaultLocale;
  if (isLocale(router.locale)) return router.locale;
  if (isLocale(router.defaultLocale)) return router.defaultLocale;
  return defaultLocale;
};

/**
 * @description This function is used to toggle the current locale.
 *
 * @param locale The current locale to be toggled.
 * @returns The new locale.
 */
export const toggleLocale = (locale: Locale) =>
  locale === locales[0] ? locales[1] : locales[0];

/**
 * @description This function is used to get the text direction of the current locale.
 *
 * @param locale The locale to be checked
 * @returns True if the locale is read right-to-left, false otherwise
 */
export const isRtl = (locale: Locale) => RTL_LOCALES.includes(locale);
