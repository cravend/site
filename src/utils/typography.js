import Typography from 'typography'

const typography = new Typography({
  baseLineHeight: 1.666,
  headerFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  bodyFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  googleFonts: [
    {
      name: 'Poppins',
      styles: ['300', '600', '700']
    }
  ]
})

export default typography
