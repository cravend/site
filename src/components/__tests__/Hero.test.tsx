import { render, screen } from "../../tests/utils";
import Hero from "../Hero";

describe("<Header />", () => {
  it("renders", () => {
    expect.assertions(1);
    const CHILD_TEXT = "Hello, world!";
    const { container } = render(
      <Hero>
        <p>{CHILD_TEXT}</p>
      </Hero>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders with a diagonal when isDiagonal is true", () => {
    expect.assertions(1);
    const CHILD_TEXT = "Hello, world!";
    const { container } = render(
      <Hero>
        <p>{CHILD_TEXT}</p>
      </Hero>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders the diagonal in the opposite direction when a language is RTL", () => {
    expect.assertions(1);
    const CHILD_TEXT = "Hello, world!";
    const { container } = render(
      <Hero>
        <p>{CHILD_TEXT}</p>
      </Hero>,
      { router: { locale: "ar" } }
    );
    expect(container).toMatchSnapshot();
  });

  it("displays children", () => {
    expect.assertions(1);
    const CHILD_TEXT = "Hello, world!";
    render(
      <Hero>
        <p>{CHILD_TEXT}</p>
      </Hero>
    );
    expect(screen.getByText(CHILD_TEXT)).toMatchSnapshot();
  });
});
