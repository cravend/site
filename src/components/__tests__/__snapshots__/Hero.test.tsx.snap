// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> displays children 1`] = `
<p>
  Hello, world!
</p>
`;

exports[`<Header /> renders 1`] = `
<div>
  <header
    class="css-10r473c"
  >
    <p>
      Hello, world!
    </p>
  </header>
</div>
`;

exports[`<Header /> renders with a diagonal when isDiagonal is true 1`] = `
<div>
  <header
    class="css-10r473c"
  >
    <p>
      Hello, world!
    </p>
  </header>
</div>
`;
